app/assets/src/components/ui/controls/dropdowns/common/DropdownTrigger.jsx

```diff
@@ -1,9 +1,56 @@
 import React from "react";
 import cx from "classnames";
+import { get } from "lodash/fp";
+
 import PropTypes from "prop-types";
 import cs from "./dropdown_trigger.scss";
 
 class DropdownTrigger extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.resizeObserver = null;
+    this.labelContainerRef = React.createRef();
+    this.labelRef = React.createRef();
+
+    this.state = {
+      hideDropdownLabel: false,
+    };
+  }
+
+  componentDidMount() {
+    const { value, hideBadgeIfInsufficientSpace } = this.props;
+
+    if (hideBadgeIfInsufficientSpace) {
+      const hasDropdownLabel = get("type.name", value) === "DropdownLabel";
+      if (
+        hasDropdownLabel &&
```

Comment:
Looks like there's a bug here.

If the Dropdown starts out with no DropdownLabel (because no options are selected), but then you select an option, the resize code will not run (and you'll get the same bug as before). This is because you have a hasDropdownLabel guard around the resize observer here, and componentDidMount only runs when the component is initialized. (componentDidUpdate is the one that runs whenever the props update)

The least complex solution (but it is slightly less performant) is to take the hasDropdownLabel guard, and move it into `handleFilterResize`. So the resizeObserver always gets registered, but the resize code only runs when a DropdownLabel is passed in.